src/main/MQ2Inlines.h

```diff
@@ -252,90 +252,394 @@ inline bool HasSkill(int nSkill)
 	return pLocalPC && pLocalPC->HasSkill(nSkill);
 }
 
-inline float GetDistance(float X1, float Y1)
-{
-	float dX = X1 - pControlledPlayer->X;
-	float dY = Y1 - pControlledPlayer->Y;
-	return sqrtf(dX * dX + dY * dY);
-}
-
+/**
+ * @brief Calculates and returns the 2D Euclidean distance between two points in a 2D space.
+ *
+ * @note If you don't need the actual distance, but only need to compare distances, you should
+ * use GetDistanceSquared instead.
+ *
+ * @param X1 The X coordinate of the first point.
+ * @param Y1 The Y coordinate of the first point.
+ * @param X2 The X coordinate of the second point.
+ * @param Y2 The Y coordinate of the second point.
+ * @return float The 2D Euclidean distance between the two points.
+ */
 inline float GetDistance(float X1, float Y1, float X2, float Y2)
 {
 	float dX = X1 - X2;
 	float dY = Y1 - Y2;
 	return sqrtf(dX * dX + dY * dY);
 }
 
-inline float GetDistanceSquared(float X1, float Y1, float X2, float Y2)
+/**
+ * @brief Calculates and returns the 3D Euclidean distance between two points in a 3D space.
+ *
+ * @note If you don't need the actual distance, but only need to compare distances, you should
+ * use GetDistanceSquared instead.
+ *
+ * @param X1 The X coordinate of the first point.
+ * @param Y1 The Y coordinate of the first point.
+ * @param Z1 The Z coordinate of the first point.
+ * @param X2 The X coordinate of the second point.
+ * @param Y2 The Y coordinate of the second point.
+ * @param Z2 The Z coordinate of the second point.
+ * @return float The 3D Euclidean distance between the two points.
+ */
+inline float GetDistance(float X1, float Y1, float Z1, float X2, float Y2, float Z2)
 {
 	float dX = X1 - X2;
 	float dY = Y1 - Y2;
-	return dX * dX + dY * dY;
+	float dZ = Z1 - Z2;
+	return sqrtf(dX * dX + dY * dY + dZ * dZ);
 }
 
+/**
+ * @brief Calculates and returns the 3D Euclidean distance between two glm::vec3 points in a 3D space.
+ *
+ * @note If you don't need the actual distance, but only need to compare distances, you should
+ * use GetDistanceSquared instead.
+ *
+ * @param first The first glm::vec3 point.
+ * @param second The second glm::vec3 point.
+ * @return float The 3D Euclidean distance between the two glm::vec3 points.
+ */
+inline float GetDistance(const glm::vec3& first, const glm::vec3& second)
+{
+	return GetDistance(first.x, first.y, first.z, second.x, second.y, second.z);
+}
+
+[[deprecated("Use GetDistance instead.")]]
```

Comment:
Use the DEPRECATE macro, for some reason these may cause issues with the clang offset generator

Reply:
Done